modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/inputs/ContentTypeInput.js

```diff
@@ -37,90 +40,239 @@ const CONTENT_TYPES = [
+							<ClayCheckbox
+								checked={!!selectedTypes.length}
+								indeterminate={
+									!!selectedTypes.length &&
+									selectedTypes.length !==
+										CONTENT_TYPES.length
+								}
+								onChange={() =>
+									setSelectedTypes(
+										!selectedTypes.length
+											? contentTypesClassNames
+											: []
+									)
+								}
+							/>
+						</ManagementToolbar.Item>
+
+						<ManagementToolbar.Item>
+							{selectedTypes.length ? (
+								<>
+									<span className="component-text">
+										{Liferay.Util.sub(
+											Liferay.Language.get(
+												'x-of-x-selected'
+											),
+											selectedTypes.length,
+											CONTENT_TYPES.length
+										)}
+									</span>
+
+									{selectedTypes.length <
+										CONTENT_TYPES.length && (
+										<ClayButton
+											displayType="link"
+											onClick={() => {
+												setSelectedTypes(
+													contentTypesClassNames
+												);
+											}}
+											size="sm"
+										>
+											{Liferay.Language.get('select-all')}
+										</ClayButton>
+									)}
+								</>
+							) : (
+								<span className="component-text">
+									{Liferay.Language.get('select-all')}
+								</span>
+							)}
+						</ManagementToolbar.Item>
+					</ManagementToolbar.ItemList>
+				</div>
+			</ManagementToolbar.Container>
+
+			<ClayModal.Body scrollable>
+				<ClayTable>
+					<ClayTable.Body>
+						{CONTENT_TYPES.map(({className, displayName}) => {
+							const isSelected = selectedTypes.includes(
+								className
+							);
+
+							return (
+								<ClayTable.Row
+									active={isSelected}
+									key={className}
+									onClick={_handleRowCheck(className)}
+								>
+									<ClayTable.Cell>
+										<ClayCheckbox
+											aria-label={Liferay.Util.sub(
+												Liferay.Language.get(
+													'select-x'
+												),
+												[displayName]
+											)}
+											checked={isSelected}
+											onChange={_handleRowCheck(
+												className
+											)}
+										/>
+									</ClayTable.Cell>
+
+									<ClayTable.Cell expanded headingTitle>
+										{displayName}
+									</ClayTable.Cell>
+								</ClayTable.Row>
+							);
+						})}
+					</ClayTable.Body>
+				</ClayTable>
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton
+							displayType="secondary"
+							onClick={_handleCancel}
+						>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+
+						<ClayButton
+							disabled={!selectedTypes.length}
+							onClick={_handleDone}
+						>
+							{Liferay.Language.get('done')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</ClayModal>
+	);
+}
+
+export default function ContentTypeInput({onBlur, onChange, touched, value}) {
+	const {observer, onOpenChange, open} = useModal();
+
+	const _getSelectedTypes = (items) => (items === '' ? [] : items.split(','));
```

Comment:
I think I'm starting to not like single line functions due to readability. Unless this is inline, I think it looks a lot more readable in this format: 

```js
const _getSelectedTypes = (items) => {
    return items === '' ? [] : items.split(',');
}
```

The ternary operator in the return makes it more to read in a single line. Plus, if we ever needed to add to the function, it makes it a bit smoother. But let me know what you think.

I think this format would be nicer but it looks like it'll get formatted into a single line.

```js
items === '' 
    ? []
    : items.split(',');
```

Reply:
Oh! Yes that definitely looks more readable, though I think I added some more commits since then (having to do with default = all selected), and it looks like:
```
	const _getSelectedTypes = (items) =>
		items
			? items.split(',')
			: CONTENT_TYPES.map(({className}) => className);
```
It got formatted to be on separate lines, should we still have it `return` too?

Reply:
I like the return because it makes sure the return statement is in a separate line and if we ever need to add more to the function, it makes it easier to add.